src/manifest/manifest_test.ts

```diff
@@ -1273,3 +1273,88 @@ Deno.test("Manifest throws error when Providers with duplicate provider_keys are
     assertStringIncludes(error.message, "OAuth2Provider");
   }
 });
+
+Deno.test("SlackManifest() oauth2 providers get set properly with token_url_config", () => {
+  const providerKey1 = "test_provider_with_token_url_config_unset";
+  const providerKey2 =
+    "test_provider_with_use_basic_authentication_scheme_false";
+  const providerKey3 =
+    "test_provider_with_use_basic_authentication_scheme_true";
+
+  const Provider1 = DefineOAuth2Provider({
+    provider_key: providerKey1,
+    provider_type: Schema.providers.oauth2.CUSTOM,
+    options: {
+      "client_id": "123.456",
+      "scope": ["scope_a", "scope_b"],
+      "token_url_config": {},
+    },
+  });
+
+  const Provider2 = DefineOAuth2Provider({
+    provider_key: providerKey2,
+    provider_type: Schema.providers.oauth2.CUSTOM,
+    options: {
+      "client_id": "123.456",
+      "scope": ["scope_a", "scope_b"],
+      "token_url_config": {
+        "use_basic_authentication_scheme": false,
+      },
+    },
+  });
+
+  const Provider3 = DefineOAuth2Provider({
+    provider_key: providerKey3,
+    provider_type: Schema.providers.oauth2.CUSTOM,
+    options: {
+      "client_id": "123.456",
+      "scope": ["scope_a", "scope_b"],
+      "token_url_config": {
+        "use_basic_authentication_scheme": true,
+      },
+    },
+  });
+
+  const definition: SlackManifestType = {
+    name: "Name",
+    description: "Description",
+    icon: "icon.png",
+    botScopes: [],
+    externalAuthProviders: [Provider1, Provider2, Provider3],
+  };
+  assertEquals(definition.externalAuthProviders, [
+    Provider1,
+    Provider2,
+    Provider3,
+  ]);
+  const Manifest = new SlackManifest(definition);
+  const exportedManifest = Manifest.export();
+
+  assertEquals(exportedManifest.external_auth_providers, {
```

Comment:
It might be good to isolate the different behaviors in separate test cases or separate test steps, this would aim to provide more context to the unit tests and help us in the future 

Deno has this concept of [test steps](https://deno.land/manual@v1.36.4/basics/testing#test-steps) that allow to create `steps` inside a test, this feels like a good use case for this.

Maybe we could have one test with 3 steps that validate the behavior of `basic_authentication_scheme` when undefined, falsy and truthy?

Reply:
I did look at Deno tests steps but was thinking, maybe this is not about testing some chronological steps but just different values.
Since you have mentioned about providing more contexts about the tests, I have included some comments. Hope this helps!
Thanks!

Reply:
This does provide more context 💯 

In the `deno-slack-api` we use this pattern for non chronological test mainly to reduce duplicate code, but for this  test the additional comments should do the trick 🥇